packages/provider/src/transactions.ts

```diff
@@ -0,0 +1,30 @@
+import { commons } from '@0xsequence/core'
+import { ethers } from 'ethers'
+
+export function validateTransactionRequest(wallet: string, transaction: commons.transaction.Transactionish) {
+  const transactions = commons.transaction.fromTransactionish(wallet, transaction)
+  const unwound = commons.transaction.unwind(wallet, transactions)
+  unwound.forEach(transaction => validateTransaction(wallet, transaction))
+}
+
+function validateTransaction(wallet: string, transaction: commons.transaction.Transaction) {
+  if (transaction.to.toLowerCase() === wallet.toLowerCase()) {
```

Comment:
ALL calls to self should fail, there is no reason for a dapp to be asking the wallet to call itself. If this is a problem we could get a bit more fancy, but for now I think this would be wise.

Reply:
I'm not sure if this is still the case, but we needed to be able to wrap some transactions in a `selfExecute` bundle in order to ensure the relayer fee was still getting paid while the internal bundle was treated as atomic:

https://github.com/horizon-games/issue-tracker/issues/4647#issuecomment-853226890

Reply:
I think we don't wan to expose this to dapps, gas refunds are handled entirely by wallet-webapp.
Besides, priority should be safety.